homeassistant/components/screenlogic/services.py

```diff
@@ -69,9 +84,58 @@ async def async_set_color_mode(service_call: ServiceCall):
             except ScreenLogicError as error:
                 raise HomeAssistantError(error) from error
 
+    async def async_set_scg(service_call: ServiceCall):
+        if not (
+            screenlogic_entry_ids := await extract_screenlogic_config_entry_ids(
+                service_call
+            )
+        ):
+            raise HomeAssistantError(
+                f"Failed to call service '{SERVICE_SET_SCG}'. Config entry for target not found"
+            )
+
+        for entry_id in screenlogic_entry_ids:
+            coordinator = hass.data[DOMAIN][entry_id]
+
+            pool_value = service_call.data.get(ATTR_POOL_PERCENT)
+            spa_value = service_call.data.get(ATTR_SPA_PERCENT)
+
+            if not pool_value and not spa_value:
+                raise HomeAssistantError(
+                    f"Failed to call service '{SERVICE_SET_SCG}'. No values specified"
+                )
+
+            _LOGGER.debug(
+                "Service '%s' called on %s with values: Pool: %s, Spa: %s",
+                SERVICE_SET_SCG,
+                coordinator.gateway.name,
+                pool_value if pool_value else "Not set",
+                spa_value if spa_value else "Not set",
+            )
+
+            if not pool_value:
+                pool_value = coordinator.data[SL_DATA.KEY_SCG]["scg_level1"]["value"]
+
+            if not spa_value:
+                spa_value = coordinator.data[SL_DATA.KEY_SCG]["scg_level2"]["value"]
```

Comment:
Since the current values are already available, why not make these number entities instead?

Reply:
TBH, I hadn't thought about that. The values exist as sensor entities currently. 
Converting them over to number entities I think should be doable, but technically a breaking change. Would that still be preferred?

Reply:
We prefer entities over services any time.

Reply:
Makes sense. 👍🏻   
I'll close this PR and re-work as number entities.